The link-swapping crashes!

When in any of the sub-functionalities (Videos for example), and trying to go back to the main menu using the arrow back (at the action bar at the top), the app crashes. Below attached log.

02-20 15:01:21.853  14293-14293/edu.yalestc.yalepublic E/AndroidRuntime﹕ FATAL EXCEPTION: main
    Process: edu.yalestc.yalepublic, PID: 14293
    java.lang.RuntimeException: Unable to start activity ComponentInfo{edu.yalestc.yalepublic/edu.yalestc.yalepublic.MainActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.String.hashCode()' on a null object reference
            at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2298)
            at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2360)
            at android.app.ActivityThread.access$800(ActivityThread.java:144)
            at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1278)
            at android.os.Handler.dispatchMessage(Handler.java:102)
            at android.os.Looper.loop(Looper.java:135)
            at android.app.ActivityThread.main(ActivityThread.java:5221)
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:372)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:899)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:694)
     Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.String.hashCode()' on a null object reference
            at edu.yalestc.yalepublic.MainActivity.onCreate(MainActivity.java:98)
            at android.app.Activity.performCreate(Activity.java:5933)
            at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1105)
            at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2251)
            at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2360)
            at android.app.ActivityThread.access$800(ActivityThread.java:144)
            at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1278)
            at android.os.Handler.dispatchMessage(Handler.java:102)
            at android.os.Looper.loop(Looper.java:135)
            at android.app.ActivityThread.main(ActivityThread.java:5221)
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:372)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:899)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:694)

--- a/src/edu/yalestc/yalepublic/Events/EventsDetails.java
+++ b/src/edu/yalestc/yalepublic/Events/EventsDetails.java
@@ -35,6 +35,9 @@ public class EventsDetails extends Activity {
 
         super.onCreate(savedInstanceState);
         extras = getIntent().getExtras();
+        if (extras == null)  // safety check
+            return;
+
         display = new DisplayMetrics();
         getWindowManager().getDefaultDisplay().getMetrics(display);
         width = display.widthPixels;

--- a/src/edu/yalestc/yalepublic/Events/EventsDisplay.java
+++ b/src/edu/yalestc/yalepublic/Events/EventsDisplay.java
@@ -21,6 +21,7 @@ public class EventsDisplay extends Activity {
     //    private Fragment dayTab;
     private Fragment monthTab;
     private Fragment listTab;
+    Bundle extras;
 
     @Override
     public boolean onCreateOptionsMenu(Menu menu) {
@@ -52,8 +53,12 @@ public class EventsDisplay extends Activity {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_events_display);
 
-        monthTab = CalendarFragment.newInstance(getIntent().getExtras());
-        listTab = ListFragment.newInstance(getIntent().getExtras());
+        extras = getIntent().getExtras();
+        if (extras == null)  // safety check
+            return;
+
+        monthTab = CalendarFragment.newInstance(extras);
+        listTab = ListFragment.newInstance(extras);
 
         //lets use the deprecated versions for now... this is deprecated only in android L
         ActionBar actionBar = getActionBar();

--- a/src/edu/yalestc/yalepublic/Events/SearchEvents/SearchByName.java
+++ b/src/edu/yalestc/yalepublic/Events/SearchEvents/SearchByName.java
@@ -36,7 +36,8 @@ public class SearchByName extends Activity {
         Intent intent = getIntent();
         if (Intent.ACTION_SEARCH.equals(intent.getAction())) {
             String query = intent.getStringExtra(SearchManager.QUERY);
-            doMySearch(query);
+            if (query != null)
+                doMySearch(query);
         }
     }
 
@@ -44,7 +45,8 @@ public class SearchByName extends Activity {
     protected void onNewIntent(Intent intent) {
         if (Intent.ACTION_SEARCH.equals(intent.getAction())) {
             String query = intent.getStringExtra(SearchManager.QUERY);
-            doMySearch(query);
+            if (query != null)
+                doMySearch(query);
         }
     }
 

--- a/src/edu/yalestc/yalepublic/ImageActivity.java
+++ b/src/edu/yalestc/yalepublic/ImageActivity.java
@@ -31,13 +31,19 @@ public class ImageActivity extends Activity {
     private Bitmap mBitmap;
     private Bitmap mBitmap2;
     private String title;
+    Bundle extras;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_image);
         ImageView imageView = (ImageView) findViewById(R.id.photoImageView);
-        photoId = getIntent().getExtras().getString(PhotoList.PHOTO_ID_KEY);
-        setTitle(getIntent().getExtras().getString(PhotosWithinAlbum.TITLE_KEY));
+        extras = getIntent().getExtras();
+        if (extras == null)  // safety check
+            return;
+
+        photoId = extras.getString(PhotoList.PHOTO_ID_KEY);
+        setTitle(extras.getString(PhotosWithinAlbum.TITLE_KEY));
         getPhotoTask task = new getPhotoTask();
         try {
             mBitmap2=task.execute().get();

--- a/src/edu/yalestc/yalepublic/MainActivity.java
+++ b/src/edu/yalestc/yalepublic/MainActivity.java
@@ -91,27 +91,30 @@ public class MainActivity extends Activity {
         actionbar.setDisplayShowTitleEnabled(true);  // Show activity title/subtitle
         actionbar.setDisplayUseLogoEnabled(false);   // Use activity logo instead of activity icon
         //actionbar.setTitle("Yale");                // Set title
-        Intent i = getIntent();
+
         String rotatingLink = getIntent().getStringExtra("url");
         ImageButton rotatingImageButton = (ImageButton) findViewById(R.id.bArts);
         TextView rotatingTextView = (TextView) findViewById(R.id.tArts);
-        switch (rotatingLink){
-            case "http://artscalendar.yale.edu" :
-                rotatingTextView.setText("Arts Calendar");
-                rotatingImageButton.setImageResource(R.drawable.module_arts_events_default);
-                break;
-            case "http://yalecollege.yale.edu/freshman" :
-                rotatingTextView.setText("Orientation");
-                rotatingImageButton.setImageResource(R.drawable.module_orientation_default);
-                break;
-            case "http://aya.yale.edu/content/current-assembly" :
-                rotatingTextView.setText("AYA Assembly");
-                rotatingImageButton.setImageResource(R.drawable.module_assembly_default);
-                break;
-            case "http://commencement.yale.edu/" :
-                rotatingTextView.setText("Commencement");
-                rotatingImageButton.setImageResource(R.drawable.module_commencement_default);
-                break;
+
+        if (rotatingLink != null) {
+            switch (rotatingLink) {
+                case "http://artscalendar.yale.edu":
+                    rotatingTextView.setText("Arts Calendar");
+                    rotatingImageButton.setImageResource(R.drawable.module_arts_events_default);
+                    break;
+                case "http://yalecollege.yale.edu/freshman":
+                    rotatingTextView.setText("Orientation");
+                    rotatingImageButton.setImageResource(R.drawable.module_orientation_default);
+                    break;
+                case "http://aya.yale.edu/content/current-assembly":
+                    rotatingTextView.setText("AYA Assembly");
+                    rotatingImageButton.setImageResource(R.drawable.module_assembly_default);
+                    break;
+                case "http://commencement.yale.edu/":
+                    rotatingTextView.setText("Commencement");
+                    rotatingImageButton.setImageResource(R.drawable.module_commencement_default);
+                    break;
+            }
         }
 
         /*Configuration config = getResources().getConfiguration();

--- a/src/edu/yalestc/yalepublic/News/NewsReader.java
+++ b/src/edu/yalestc/yalepublic/News/NewsReader.java
@@ -63,7 +63,7 @@ public class NewsReader extends Activity {
         Log.d("NewsReader passed", downloadurl);
 
         // If we're online, downloads the RSS Feed and returns it as `feed`
-        if (isOnline()) {
+        if (isOnline() && downloadurl != null) {
             NewsDownload start = new NewsDownload();
             try {
                 feed = start.execute(downloadurl).get();

--- a/src/edu/yalestc/yalepublic/VideoYoutubePlayback.java
+++ b/src/edu/yalestc/yalepublic/VideoYoutubePlayback.java
@@ -52,7 +52,7 @@ public class VideoYoutubePlayback extends YouTubeBaseActivity implements YouTube
 
 
         /** Start buffering **/
-        if (!wasRestored) {
+        if (!wasRestored && VIDEO_ID != null) {
             player.cueVideo(VIDEO_ID);
         }
     }

--- a/src/edu/yalestc/yalepublic/Videos/VideoYoutubePlayback.java
+++ b/src/edu/yalestc/yalepublic/Videos/VideoYoutubePlayback.java
@@ -52,7 +52,7 @@ public class VideoYoutubePlayback extends YouTubeBaseActivity implements YouTube
 
 
         /** Start buffering **/
-        if (!wasRestored) {
+        if (!wasRestored && VIDEO_ID != null) {
             player.cueVideo(VIDEO_ID);
         }
     }

--- a/src/edu/yalestc/yalepublic/Videos/VideosWithinPlaylist.java
+++ b/src/edu/yalestc/yalepublic/Videos/VideosWithinPlaylist.java
@@ -34,7 +34,7 @@ public class VideosWithinPlaylist extends Activity {
         //to get the passed parameters
         extras = getIntent().getExtras();
         setContentView(R.layout.activity_video_within_list);
-        if (savedInstanceState == null) {
+        if (savedInstanceState == null && extras != null) {
             getFragmentManager().beginTransaction()
                     .add(R.id.container1, PlaceholderFragment.newInstance(extras)).commit();
         }

--- a/src/edu/yalestc/yalepublic/VideosWithinPlaylist.java
+++ b/src/edu/yalestc/yalepublic/VideosWithinPlaylist.java
@@ -58,6 +58,8 @@ public class VideosWithinPlaylist extends Activity {
         //to get the passed parameters
         context = this;
         extras = getIntent().getExtras();
+        if (extras == null)  // safety check
+            return;
         setContentView(R.layout.activity_photo_within_album);
         if (savedInstanceState == null) {
             getFragmentManager().beginTransaction()

